File: ls.php
#!/usr/bin/php

<?

 $file = popen("ls -aFl ~", "r");
 $read = fread($file, 2096);
# echo $read;
 $dir = 0;
 $file_num = 0;
 $result = "";
 $line = explode("\n", $read);
 $spl = " ";
 for($i=1; $i<count($line)-1; $i++)
 {	 

	$word1 = preg_split("/[\s]+/",$line[$i]); 
	if(substr($word1[8], 0, 2) == './')
	{
		
			$dir++;
	}
	elseif(substr($word1[8], 0, 3) == '../')
	{
		$dir++;
	}
	else{
		if(substr($word1[0], 0, 1) == 'd')
			$dir++;
		else
			$file_num++;
		echo "$word1[8] $word1[4] $word1[0]\n";
	}

	

	
 }

 echo $dir . " " . $file_num. "\n";
 echo $result;


 pclose($file);


?>

File: ls.py
#!usr/bin/python3
import subprocess
import sys
import re

var1 = sys.argv[1]
temp = 'ls -aFl ' + var1

ls = subprocess.Popen(temp,shell = True, stdout=subprocess.PIPE).communicate()[0].decode('utf-8').strip()

ls1 = []
directory = 0
file_num = 0
result = []
ls1 = ls.splitlines()
for i in ls1:
    
    if i.endswith('./'):
        directory = directory + 1

    elif i.startswith('d'):
        directory = directory + 1
        result.append(i)
    elif i.startswith('t'):
        print("")
    else:
        file_num = file_num + 1
        result.append(i)

result1 = ""

print("directory = %d, file %d" %(directory, file_num))
for item in result:
    item1 = re.split("\s+", item)
    print(item1[8],item1[4],item1[0])

